src/index.ts

```diff
@@ -19,12 +18,28 @@ export function encode(input: Input): Buffer {
     }
     const buf = Buffer.concat(output)
     return Buffer.concat([encodeLength(buf.length, 192), buf])
-  } else {
-    const inputBuf = toBuffer(input)
-    return inputBuf.length === 1 && inputBuf[0] < 128
-      ? inputBuf
-      : Buffer.concat([encodeLength(inputBuf.length, 128), inputBuf])
   }
+
+  // Edge case, see https://eth.wiki/en/fundamentals/rlp
+  // "The integer 0 = [ 0x80 ]"
+  // EthereumJS-util would convert 0 to the Buffer.from('00', 'hex'), which would
+  // thus return Buffer.from('00', 'hex') when encoding that: this does not match the spec.
+  if (input === 0 || input === BigInt(0)) {
+    return Buffer.from('80', 'hex')
+  }
```

Comment:
suggestion to rewrite the comment more succinctly

also adding the BN case I suggested in chat

(might need lint formatting)

could it be an alternative to do this check *after* `toBuffer`? `if (inputBuf.equals(Buffer.from('00', 'hex')) { return Buffer.from('80', 'hex') }`? then we don't have to check the varying input types and can rely on `toBuffer` to handle the input conversion for us and just check if the output matches `Buffer.from('00', 'hex')`

```suggestion
  // According to RLP spec (https://eth.wiki/en/fundamentals/rlp):
  // "The integer 0 = [ 0x80 ]"
  // util's toBuffer converts 0 to Buffer.from('00', 'hex')
  // so here we will return early to match the spec
  if (input === 0 || input === BigInt(0) || (BN.isBN(input) && input.isZero())) {
    return Buffer.from('80', 'hex')
  }
```